roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -287,280 +276,232 @@ def __init__(self, filepath, optical_channel_name=None,
             self.property_vals.append(np.array(ps[i].data))
 
         #Extracting stores images as GrayscaleImages:
-        self._greyscaleimages = [_nwbchildren_name[f] for f, u in enumerate(_nwbchildren_type) if u == 'GrayscaleImage']
-
-    @property
-    def image_dims(self):
-        return list(self.extimage_dims)
+        _greyscaleimages = [i for i in nwbfile.all_children() if type(i).__name__ == 'GrayscaleImage']
+        self._image_correlation = [i.data[()] for i in _greyscaleimages if 'corr' in i.name.lower()][0]
+        self._image_mean = [i.data[()] for i in _greyscaleimages if 'mean' in i.name.lower()][0]
 
-    @property
-    def no_rois(self):
-        return self.roi_idx.size
-
-    @property
-    def roi_idx(self):
-        return self._roi_idx
+    def __del__(self):
+        self._io.close()
 
-    @property
-    def accepted_list(self):
+    def get_accepted_list(self):
         if self._accepted_list is None:
-            return list(range(self.no_rois))
+            return list(range(self.get_num_rois()))
```

Comment:
we shouldn't assume ROIs are accepted

Reply:
will create a separate PR to have these changes for all classes.